Source/CNPGridMenu.m

```diff
@@ -107,7 +108,27 @@ - (void)viewWillAppear:(BOOL)animated {
     self.collectionView.backgroundView.userInteractionEnabled = YES;
     [self.collectionView.backgroundView addGestureRecognizer:self.backgroundTapGestureRecognizer];
 }
+-(void)viewDidAppear:(BOOL)animated{
+    [super viewDidAppear:animated];
+    _showButtonIndex=0;
+    [self fadeInNextButton];
+}
+//If feature is enable it will fade in buttons one by one
+-(void)fadeInNextButton{
+    if(_fadeInAnimationEnabled){
+        if (_showButtonIndex<=[_menuItems count]-1) {
+            NSIndexPath *indexPath=[NSIndexPath indexPathForRow:_showButtonIndex inSection:0];
```

Comment:
Rather than applying this here, you can animate each cell in the collectionView method for [willDisplayCell](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UICollectionViewDelegate_protocol/index.html#//apple_ref/occ/intfm/UICollectionViewDelegate/collectionView:willDisplayCell:forItemAtIndexPath:)

You can have them fade in sequence by applying a delay multiplied by the indexPath of the item.

Also, make the animation time customizable.